go/internal/controller/translator/adk_api_translator.go

```diff
@@ -463,12 +463,14 @@ func (a *adkApiTranslator) translateInlineAgent(ctx context.Context, agent *v1al
 		}))
 	}
 
-	toolsByServer := make(map[v1alpha2.TypedLocalReference][]string)
 	for _, tool := range agent.Spec.Declarative.Tools {
 		// Skip tools that are not applicable to the model provider
 		switch {
 		case tool.McpServer != nil:
-			toolsByServer[tool.McpServer.TypedLocalReference] = append(toolsByServer[tool.McpServer.TypedLocalReference], tool.McpServer.ToolNames...)
+			err := a.translateMCPServerTarget(ctx, cfg, agent.Namespace, tool.McpServer, tool.HeadersFrom)
```

Comment:
Dropping the `toolsByServer` map does mean that some configuration may end up being duplicated if folks (and the UI) don't end up grouping tools per server nicely. Things do still seem to work though, and I don't see any way around this. Thoughts?

Reply:
I think you're right, the purpose originally was to try and avoid duplicated resolution of headers, but now that it can come from multiple sources that's no longer possible.